### PR TITLE
SUPERSEDED: Iso/Interrupt streaming changes to DPI

### PR DESCRIPTION
**Pull request by alees24**
Building on PR #18419 -

- Corresponding DPI changes are introduced to handle the absence of ACK/NAK responses on Isochronous streams, and to correct buffer management faults that were exposed by the increased/modified traffic. The number of buffers is increased to cope with the additional load.

Part of #18305, completing the changes to the device and DPI ends of the streams, without adding new tests or altering the behavior of existing tests.

### CODE REVIEW

**Review by marnovandermaas — COMMENTED** [review]
You mention two other PRs. What's the order in which these should be merged? Or does it not matter?
Also, do you have any instructions on how to test this code?

File: hw/dv/dpi/usbdpi/usbdpi.c [hunk 1]
```diff
@@ -195,13 +195,14 @@ void usbdpi_device_to_host(void *ctx_void, const svBitVecVal *usb_d2p) {
...
         // Nothing to do
-        ctx->state = ST_GET;
         break;
 
       default:
         assert(!"Invalid/unknown state");
         break;
     }
+
+    ctx->state = ST_GET;
```
Comment (a-will) [inline hw/dv/dpi/usbdpi/usbdpi.c:205]: Just checking, but did the change from [83597373cc5db85bc3c527b95b24ecde3a1015ab](https://github.com/lowRISC/opentitan/commit/83597373cc5db85bc3c527b95b24ecde3a1015ab#diff-953b4db380c5ee57046af37ad5d330efb1d3998b8a56d069a3538c756977a49bR198) not work out?

**Review by a-will — APPROVED** [review]
This was only a quick pass, but LGTM.

File: hw/dv/dpi/usbdpi/usbdpi.c (hunk 1 above)
Comment (alees24) [reply hw/dv/dpi/usbdpi/usbdpi.c:205]: I'm afraid it's more a case of too many branches, PRs and tests and confusion with a merge. I shall change it to what we've been using with the ASIC t-l smoke test and generated byte enables and check that the Verilator t-l sims are unaffected still; changing state to ST_GET when it was not ST_IDLE would only assist in trying to stumble on after an error that should not have occurred, but I should perhaps have left myself a comment to remind myself of that!

Comment (alees24) [reply hw/dv/dpi/usbdpi/usbdpi.c:205]: Restored the previous change and left myself/posterity a comment for next time!

**Pull request closed without merging**